### ISSUE TITLE
oscap-chroot reports "unknown" for every entry

### ISSUE THREAD

**Issue by erickeller** [OpenSCAP/openscap#736]
I would like to test an older version of Debian Jessie in the hope of finding some vulnerabilities

I was able to build the tool under Ubuntu 16.04 successfully with the following instructions:
https://gist.github.com/erickeller/e1b1229d6f69d08ba760fad4a7222dc4 

I created my chroot environement using the following command:

```
sudo debootstrap --no-check-gpg --arch="amd64" --variant=minbase --include="ssh" "jessie" "/tmp/oschroot" "http://snapshot.debian.org/archive/debian/20160508T155948Z"
```

Then I downloaded the oval definition from debian security

```
wget https://www.debian.org/security/oval/oval-definitions-2017.xml -O /tmp/oval-definitions-2017.xml
```
I hopped to encounter some security issues when running the oscap-chroot command:

```
sudo oscap-chroot /tmp/oschroot oval eval --results /tmp/jessie.xml --report /tmp/report-jessie.html /tmp/oval-definitions-2017.xml
```

The only kind of output I get are:

```
...
Definition oval:org.debian:def:20160617: unknown
Definition oval:org.debian:def:20160616: unknown
...
```

When running the scan within the chroot I get other results:

```
sudo chroot /tmp/oschroot
apt-get update && apt-get install -y libopenscap8 wget
wget https://www.debian.org/security/oval/oval-definitions-2017.xml -O /tmp/oval-definitions-2017.xml
oscap oval eval --results /tmp/jessie.xml --report /tmp/report-jessie.html /tmp/oval-definitions-2017.xml
```

```
...
Definition oval:org.debian:def:20160617: false
Definition oval:org.debian:def:20160616: false
...
```

I do not really know what's going wrong with the `oscap-chroot` I tried to get more verbose without success.
I cannot tell if the oval from the Debian security matches the requirements for the oscap tool...

Would be glad to get some help guidance here.

**Comment by mpreisler** [comment]
The dpkg probe in OpenSCAP doesn't support offline mode scanning. Probably wouldn't be difficult to implement.

**Comment by erickeller** [comment]
ok could you point out the source which handle this probe so I could have a look?

**Comment by mpreisler** [comment]
@erickeller https://github.com/OpenSCAP/openscap/tree/maint-1.2/src/OVAL/probes/unix/linux

Check out dpkg* in that folder.

See the rpm probes for examples of how to implement offline scanning. https://github.com/OpenSCAP/openscap/blob/maint-1.2/src/OVAL/probes/unix/linux/rpminfo.c#L272

The offline chroot mode might be easier for dpkg because openscap will even chroot for you.

**Comment by erickeller** [comment]
thanks for the information, I have to dig a bit more into the code, and have an overview how it works.
Is there some developer documentation describing the architecture?

I did a quick implementation adding the `probe_setoption` to the dpkginfo.c file and was able to compile and run the upper test resulting in many false and still some unknown, in fact some improvement of the 
behavior. 

Still encountering some warning about a probe_uname, `W: probe_uname: Requested offline mode is not supported by probe_uname.`

I also bumped into an older issue you answered: https://github.com/OpenSCAP/openscap/issues/678 seems some people already wanted that to be fixed.

Cheers
